fix endpoints norm for m=0 in assoc_legendre_p

fbourgey opened the pull request:
Fixes endpoints normalization error in https://github.com/scipy/scipy/issues/24099

fbourgey commented on include/xsf/legendre.h:
```diff
@@ -303,6 +303,10 @@ void assoc_legendre_p_pm1(NormPolicy norm, int n, int m, T z, int branch_cut, T
             res = T(1);
+            // Apply normalization for m=0
+            if (std::is_same_v<NormPolicy, assoc_legendre_norm_policy>) {
+                res *= sqrt(T(2 * n + 1) / T(2));
```
@jorenham I believe this is not equivalent. In templated numeric code, we should avoid fractional literals to introduce unexpected behavior.

fbourgey commented:
@steppi @lucascolley I've added a test in https://github.com/scipy/scipy/pull/24108 already. Do we need one here, too?

steppi commented:
> @steppi @lucascolley I've added a test in [scipy/scipy#24108](https://github.com/scipy/scipy/pull/24108) already. Do we need one here, too?

Yes, I think that would be nice to have.

fbourgey commented:
> > @steppi @lucascolley I've added a test in [scipy/scipy#24108](https://github.com/scipy/scipy/pull/24108) already. Do we need one here, too?
> 
> Yes, I think that would be nice to have.

@steppi, I have added a test.

steppi reviewed (approved):
Looks good. Thanks @fbourgey!
